js/ui/grid_core/ui.grid_core.data_source_adapter.js

```diff
@@ -449,7 +449,9 @@ export default gridCore.Controller.inherit((function() {
         },
         _handleCustomizeStoreLoadOptions(options) {
             this._handleDataLoading(options);
-            options.data = getPageDataFromCache(options, true) || options.cachedStoreData;
+            if(!(options.data?.data?.length === 0)) {
+                options.data = getPageDataFromCache(options, true) || options.cachedStoreData;
+            }
```

Comment:
getPageDataFromCache returns an array of items. Why the options.data property should be an object?

Reply:
Can we simplify condition to `if(!options.data)`?

Reply:
@NickMitrokhin, it seems that `getPageDataFromCache` correctly resolves when `totalCount` is required (or is called only when it is not required)

@vconst, several tests in `dataController` failed if I use `if(!options.data)`. I didn't research the reasons